worlds/rogue_legacy/__init__.py

```diff
@@ -49,6 +49,30 @@ def fill_slot_data(self) -> dict:
         return {option_name: self.get_setting(option_name).value for option_name in rl_options}
 
     def generate_early(self):
+        location_ids_used_per_game = {
+            world.game: set(world.location_id_to_name) for world in self.multiworld.worlds.values()
+        }
+        item_ids_used_per_game = {
+            world.game: set(world.item_id_to_name) for world in self.multiworld.worlds.values()
+        }
+        overlapping_games = set()
+
+        for id_lookup in (location_ids_used_per_game, item_ids_used_per_game):
+            for game_1, ids_1 in id_lookup.items():
+                for game_2, ids_2 in id_lookup.items():
+                    if game_1 == game_2:
```

Comment:
This might be out of scope for this, but consider the case, that also happens sometimes, where a single game has an overlapping ID. It is almost always a bug on that game's side, but I have encountered in the past problems in Stardew (and, presumably any multiclient game before the overlapping ID fix was introduced) where another game having a dupe ID would crash my own game. Presumably, RL is also at risk of that.

When this happens, the item/location itself might be broken, but the rest of the multiworld usually works as intended. Except if a client, like Stardew back then or RL now, crashes upon seeing that ID and then becomes unplayable.